### PR TITLE
Mesa

### PR DESCRIPTION
**Pull request by ashuntu**
- Added Mesa from https://github.com/AlanGriffiths/kisak-core20
- Fix LD cache issue where some games overrode `LD_LIBRARY_PATH`

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -31,14 +31,14 @@ layout:
     bind: $SNAP/usr/lib/i386-linux-gnu/dri
```
Comment (jhenstridge) [inline snap/snapcraft.yaml:31]: I think you want to bind `$SNAP/graphics/dri-i386` here.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -31,14 +31,14 @@ layout:
...
   /usr/lib/x86_64-linux-gnu/dri:
     bind: $SNAP/usr/lib/x86_64-linux-gnu/dri
```
Comment (jhenstridge) [inline snap/snapcraft.yaml:33]: ... and `$SNAP/graphics/dri` here.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -210,7 +243,10 @@ apps:
       TMPDIR: $XDG_RUNTIME_DIR
       ALWAYS_USE_PULSEAUDIO: 1
       DISABLE_WAYLAND: 1
-
+      LD_LIBRARY_PATH: $LD_LIBRARY_PATH:$SNAP/graphics/lib
+      LIBGL_DRIVERS_PATH: $LIBGL_DRIVERS_PATH:$SNAP/graphics/dri
+      LIBVA_DRIVERS_PATH: $LIBVA_DRIVERS_PATH:$SNAP/graphics/dri
+      __EGL_VENDOR_LIBRARY_DIRS: $__EGL_VENDOR_LIBRARY_DIRS:$SNAP/graphics/glvnd/egl_vendor.d
```
Comment (jhenstridge) [inline snap/snapcraft.yaml:249]: If we're using layouts to bind these to their usual paths under `/usr`, then maybe the environment variables aren't needed?

**Review by jhenstridge — COMMENTED** [review]
See comments below based on discussion earlier.

I've also noticed that the `mesa-core20` snap doesn't seem to include `libGLX_mesa.so.0`, which seems like an odd exclusion. I don't think we'd want to be mixing the GLX from the focal archive with the rest of the Mesa libraries from Alan's PPA. We should see if we can get Alan to include it in his snap.

**Comment by ashuntu** [comment]
Closing this PR for now since we went a different direction for Mesa.

**Pull request closed without merging**